src/shared/Core/EnvironmentBase.cs

```diff
@@ -141,6 +150,9 @@ internal virtual bool TryLocateExecutable(string program, ICollection<string> pa
             path = null;
             return false;
         }
+
+        public abstract void SetEnvironmentVariable(string variable, string value,
+            EnvironmentVariableTarget target = EnvironmentVariableTarget.Process);
```

Comment:
Both of the implementations of `SetEnvironmentVariable` (POSIX and Windows) are the same. We should just move the implementation to this `EnvironmentBase` class instead.

Will need to move `GetCurrentVariables` to a `protected abstract` method on `EnvironmentBase`, and then update the constructors that are calling this statically and replace the implementation with a `null -> GetCurrentVariables()` call.

E.g.

```diff
-public WindowsEnvironment(IFileSystem fileSystem)
-    : this(fileSystem, GetCurrentVariables()) { }
-
-internal WindowsEnvironment(IFileSystem fileSystem, IReadOnlyDictionary<string, string> variables)
+internal WindowsEnvironment(IFileSystem fileSystem, IReadOnlyDictionary<string, string> variables = null)
    : base(fileSystem)
{
-    EnsureArgument.NotNull(variables, nameof(variables));
-    Variables = variables;
+    Variables = variables ?? GetCurrentVariables();
}
```